modules/components/org.wso2.carbon.appfactory.core/src/main/java/org/wso2/carbon/appfactory/core/dao/JDBCResourceDAO.java

```diff
@@ -376,4 +381,241 @@ public boolean updateResource(String applicationKey, String resourceType, String
+     *
+     * @param containerList   list of containers
+     * @param deployementName name of the deployement
+     * @param connection      database connection
+     */
+    private void addContainer(List<Container> containerList, String deployementName, Connection connection) throws SQLException{
+        PreparedStatement addContainerPreparedStatement = null;
+        try {
+            addContainerPreparedStatement = connection.prepareStatement(SQLConstants.ADD_CONTAINERS);
+            for (Container container : containerList) {
+                addContainerPreparedStatement.setString(1, container.getBaseImageName());
+                addContainerPreparedStatement.setString(2, container.getBaseImageVersion());
+                addContainerPreparedStatement.setString(3, deployementName);
+                addContainerPreparedStatement.executeUpdate();
+
+                List<ServiceProxy> serviceProxyList = container.getServiceProxies();
+                addServiceProxy(serviceProxyList, container, deployementName, connection);
+            }
+        } catch (SQLException e) {
+            String message = "Error while adding the database for container to deployment : " + deployementName;
+            log.error(message, e);
+            throw new SQLException(message, e);
+        } finally {
+            AppFactoryDBUtil.closePreparedStatement(addContainerPreparedStatement);
+        }
+
+    }
+
+    /**
+     * Add service proxies to container
+     *
+     * @param serviceProxyList list of service proxies
+     * @param container        container details
+     * @param deploymentName   name of the deployment
+     * @param connection       database connection
+     */
+    private void addServiceProxy(List<ServiceProxy> serviceProxyList, Container container, String deploymentName,
+            Connection connection) throws SQLException {
+        PreparedStatement addProxiesPreparedStatement = null;
+        try {
+            addProxiesPreparedStatement = connection.prepareStatement(SQLConstants.ADD_SERVICE_PROXIES);
+            for (ServiceProxy serviceProxy : serviceProxyList) {
+                addProxiesPreparedStatement.setString(1, serviceProxy.getServiceName());
+                addProxiesPreparedStatement.setString(2, serviceProxy.getServiceProtocol());
+                addProxiesPreparedStatement.setInt(3, serviceProxy.getServicePort());
+                addProxiesPreparedStatement.setInt(4, serviceProxy.getServiceBackendPort());
+                addProxiesPreparedStatement.setString(5, container.getBaseImageName());
+                addProxiesPreparedStatement.setString(6, container.getBaseImageVersion());
+                addProxiesPreparedStatement.setString(7, deploymentName);
+                addProxiesPreparedStatement.executeUpdate();
+            }
+        } catch (SQLException e) {
+            String message = "Error while adding the database for service proxies to deployement : " + deploymentName;
```

Comment:
This is fixed in following commit.
https://github.com/wso2/product-af/pull/800/commits/b54500739fe65b9908eda6176ecbace4b76a972d